lib/util/cache.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2022 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package util
+
+import "sync"
+
+type Cache[K comparable, V any] struct {
```

Comment:
This is really just a worse `sync.Map`, an actual cache would have some max size or invalidation logic etc. Better to use a normal map with a mutex at the call site, or a sync.Map, so that the behavior is clear.

Reply:
Made some improvements